x/application/keeper/msg_server_undelegate_from_gateway_test.go

```diff
@@ -272,3 +275,249 @@ func TestMsgServer_UndelegateFromGateway_SuccessfullyUndelegateFromUnstakedGatew
 	require.Equal(t, appAddr, foundApp.Address)
 	require.Equal(t, 0, len(foundApp.DelegateeGatewayAddresses))
 }
+
+func TestMsgServer_UndelegateFromGateway_DelegationIsActiveUntilNextSession(t *testing.T) {
```

Comment:
I think a comment here with numbered steps explaining what's happening would be helpful.